scipy/interpolate/_fitpack_impl.py

```diff
@@ -654,7 +654,13 @@ def splint(a, b, tck, full_output=0):
         return list(map(lambda c, a=a, b=b, t=t, k=k:
                         splint(a, b, [t, c, k]), c))
     else:
-        aint, wrk = _fitpack._splint(t, c, k, a, b)
+        if t.size != c.size:
+            if c.size >= t.size - k - 1:
+                # If t and c are not the same size, c is padded with zeros.
+                c = np.pad(c, (0, t.size - c.size))
+            else:
+                raise ValueError("t and c are invalid length.")
+        aint, wrk = dfitpack.splint(t, c, k, a, b)
```

Comment:
This is a downside, quite a bit of extra code.

Reply:
It rather embarassingly took me a while to realize that Fortran does not really allow variable-sized arrays (unless `allocatable`), so the f2py declaration `real*8 dimension(n)` does not allow for arrays of size less than `n`. So every array needs to be accompanied by its integer size variable, and an f2py check can track if `len(c) >= len(t) - k - 1`, as in gh-17038.